src/Hl7.Fhir.Core/Rest/BaseFhirClient.cs

```diff
@@ -1111,12 +1114,41 @@ private void verifyServerVersion()
             {
                 throw Error.NotSupported($"This CapabilityStatement of the server doesn't state its FHIR version");
             }
-            else if (!ModelInfo.CheckMinorVersionCompatibility(serverVersion))
+            else if (!CheckMinorVersionCompatibility(_fhirVersion, serverVersion))
             {
-                throw Error.NotSupported($"This client supports FHIR version {ModelInfo.Version} but the server uses version {serverVersion}");
+                throw Error.NotSupported($"This client supports FHIR version {_fhirVersion} but the server uses version {serverVersion}");
             }
 
         }
+
+        private static bool CheckMinorVersionCompatibility(string version, string externalVersion)
```

Comment:
I would not mind keeping this function as a public utility method (somewhere outside the current ModelInfo of course).

Reply:
I used `SemVersion` now for this functionality.